mlflow/models/model.py

```diff
@@ -42,7 +42,6 @@ def __init__(
         flavors=None,
         signature=None,  # ModelSignature
         saved_input_example_info: Dict[str, Any] = None,
-        model_uuid=None,
```

Comment:
Instead of removing `model_uuid` from the constructor and modifying the attribute during `load()`, can we define a third type of value for `model_uuid`?

1. By default, `model_uuid` should be a function that generates a UUID, e.g. `lambda: uuid.uuid4().hex`
2. Model uuid could be `None`, indicating that the model has no ID
3. Model uuid could be a string, indicating that the model already has a UUID

In the constructor, we can check if the input is a function and, if it is, call it to generate an ID. Otherwise, set `self.model_uuid = model_uuid`.

Reply:
sounds good